### PR TITLE
[IMP] runbot_send_email: Added unit tests to increase the coverage

### PR DESCRIPTION
**Pull request by luiseevaquer**
- [x] Add integration with mailgun sandbox to send tests email
- [x] Add .travis.yml file with encrypt variables env (secure: https://docs.travis-ci.com/user/encryption-keys/) 

https://github.com/Vauxoo/runbot-addons/issues/70

PS:
test_done

### CODE REVIEW

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            'repo_id': self.repo_obj.id,
+            'name': branch_name,
+            'branch_name': 'fast-travis',
+            'sticky': False,
+            'coverage': False,
+        })
+        recipient = 'testcommitter@runbot.vauxoo.com'
+        self.build_obj = self.env['runbot.build'].create({
+            'branch_id': self.branch_obj.id,
+            'name': 'fcb98eb195fc62fa49873f8f101f1738e38ea7c0',
```
Comment (nhomar) [inline runbot_send_email/tests/test_runbot_send_email.py:46]: mosca esto no debería estar aquí borralo y haz rebase pliz.

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        domain = [ ('repo_id','=',self.repo_obj.id), ]
+        self.build = self.build_obj.search(domain)
+        if os.environ.get('EMAIL_PASSWORD', False):
+            setattr(threading.currentThread(), 'testing', False)
+            self.build.pool._init = False
+            self.mail_server = self.env['ir.mail_server'].create({
+                'smtp_port': '25',
+                'smtp_host': 'smtp.mailgun.org',
+                'smtp_encryption': 'starttls',
+                'smtp_user': 'postmaster@runbot.vauxoo.com',
```
Comment (nhomar) [inline runbot_send_email/tests/test_runbot_send_email.py]: igual.. usa el usuario sandbox, no postmaster.

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        if os.environ.get('EMAIL_PASSWORD'):
+            with self.assertRaisesRegexp( exceptions.except_orm,
+                                         'Connection Test Succeeded!'):
+                self.mail_server.test_smtp_connection()
+
+    def test_20_send_email(self):
+        """
+        This test tries to send a runbot email with password
+        """
+        if os.environ.get('EMAIL_PASSWORD'):
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: We should get the tests without a password...
Check the main example: https://github.com/Vauxoo/yoytec/blob/8.0/mail_notification_collection/tests/test_template_collections.py#L74-L78

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    This create a runbot send email test
+    """
+
+    def setUp(self):
+        """
+        This add required environment variable for test
+        """
+        super(TestRunbotSendEmail, self).setUp()
+        repo_name = 'https://github.com/vauxoo-dev/'
+        repo_name += 'runbot_branch_remote_name_grp_feature2.git'
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: Do you need a real repository url to execute `github_status` method?

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            'committer_email': recipient,
+            'subject': '[TEST] Test message commit',
+            'date': '2016-03-08 00:00:00',
+       })
+        domain = [ ('repo_id','=',self.repo_obj.id), ]
+        self.build = self.build_obj.search(domain)
+        if os.environ.get('EMAIL_PASSWORD', False):
+            setattr(threading.currentThread(), 'testing', False)
+            self.build.pool._init = False
+            self.mail_server = self.env['ir.mail_server'].create({
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: You have this record in data demo.
What about extract it with ref('xml_id') and write the password value?

Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: Check the example: https://github.com/Vauxoo/yoytec/blob/8.0/mail_notification_collection/demo/ir_mail_server_demo.xml

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        recipient = 'testcommitter@runbot.vauxoo.com'
+        self.build_obj = self.env['runbot.build'].create({
+            'branch_id': self.branch_obj.id,
+            'name': 'fcb98eb195fc62fa49873f8f101f1738e38ea7c0',
+            'author': 'Test Author',
+            'author_email': 'author@runbot.vauxoo.com',
+            'committer': 'Test Committer',
+            'committer_email': recipient,
+            'subject': '[TEST] Test message commit',
+            'date': '2016-03-08 00:00:00',
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py:52]: Use current_date here, because runbot skip some builds with old date.

**Comment by moylop260** [comment]
:+1: 
The coverage now is of 90%

Could you add a print-screen of inbox email received with the env-var?

After merge this one, I will add a travis-encrypt env-var

**Comment by moylop260** [comment]
The red of travis is unrelated

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            'repo_id': self.repo_obj.id,
+            'name': branch_name,
+            'branch_name': 'branch_name',
+            'sticky': False,
+            'coverage': False,
+        })
+        recipient = 'committer@testsend.com'
+        self.build_obj = self.env['runbot.build'].create({
+            'branch_id': self.branch_obj.id,
+            'name': 'fcb98eb195fc62fa49873f8f101f1738e38ea7c0',
```
Comment (nhomar) [inline runbot_send_email/tests/test_runbot_send_email.py:46]: sigue estando la clave aqui.

Comment (nhomar) [inline runbot_send_email/tests/test_runbot_send_email.py:46]: olvidaleo

File: runbot_send_email/demo/ir_mail_server_demo.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<openerp>
+    <data>
+        <record id="runbot_send_ir_mail_server_demo" model="ir.mail_server">
+            <field name="name">Runbot Email Server (TEST)</field>
+            <field name="sequence">5</field>
+            <field name="smtp_host">smtp.mailgun.org</field>
+            <field name="smtp_port">25</field>
+            <field name="smtp_encryption">starttls</field>
+            <field name="smtp_user">postmaster@sandbox284670b593e244c381b05093de353387.mailgun.org</field>
```
Comment (nhomar) [inline runbot_send_email/demo/ir_mail_server_demo.xml]: ESTO no debe estar acá! esto es privado.

Comment (moylop260) [inline runbot_send_email/demo/ir_mail_server_demo.xml]: @lescobarvx then we can use other env-var-encrypt for this value too.

**Comment by moylop260** [comment]
travis red unrelated.

**Comment by luiseevaquer** [comment]
coverage_100
local coverage

**Pull request merged**